include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -702,47 +702,43 @@ __parallel_radix_sort(_ExecutionPolicy&& __exec, _Range&& __in_rng)
 #if __SYCL_COMPILER_VERSION >= 20230101 //for Intel(R) oneAPI C++ Compiler Classic 2023 and later
     //TODO: 1.to reduce number of the kernels; 2.to define work group size in runtime, depending on number of elements
     constexpr auto __wg_size = 64;
-    if (__n <= 32768 && __wg_size * 8 <= __max_wg_size)
+    if (__n <= 32768 && __wg_size * 16 <= __max_wg_size)
```

Comment:
What is the reason to change work_group size?

Reply:
to use one group radix sort for 16K-32K as well.
It gives better result as the general  pattern.

Reply:
I see that 16K-32K case is was supported as well. The change doubles work-group size and halves block size (aka number of elements per work-item).

Reply:
I have a concern about that condition `if (__n <= 32768 && __wg_size * 16 <= __max_wg_size)` and to be more precise about `__wg_size * 16 <= __max_wg_size` part. You limit usage of single-work-group algorithm to devices with max work-group size 1024 and more. However, it is only needed for 16k-32k case.